src/julia.h

```diff
@@ -289,19 +289,21 @@ typedef struct _jl_code_info_t {
         // 1 << 11 = :inaccessiblemem_or_argmemonly
         // 1 << 12-19 = callsite effects overrides
     // miscellaneous data:
-    jl_value_t *method_for_inference_limit_heuristics; // optional method used during inference
     jl_value_t *linetable; // Table of locations [TODO: make this volatile like slotnames]
     jl_array_t *slotnames; // names of local variables
     jl_array_t *slotflags;  // local var bit flags
     // the following are optional transient properties (not preserved by compression--as they typically get stored elsewhere):
     jl_value_t *slottypes; // inferred types of slots
-    jl_value_t *rettype;
-    jl_method_instance_t *parent; // context (optionally, if available, otherwise nothing)
```

Comment:
Would it be horrible if I asked if we could put this field back? It seems to be necessary to be able to do decent `show` code, as evidenced by the more complicated code required in `InteractiveUtils.code_warntype` to recover this functionality. Even if we don't use it anywhere else, based on the current design of `code_typed`, it had seemed like this duplication was worth the trade off (better code printing ability vs. an deleting an otherwise unused field that exists only when uncompressed)

Reply:
I think it would be ok to put it back, but if we put it back, the place that sets it should be code_typed. However, I would like to do that in a self-contained follow-up PR. I would also maybe like to name the field something else so that people don't accidentally assume it's always set.

Reply:
I think we are only one-two bugfixes away from being able to guarantee this is field is always available (I have experimented with that locally)

Reply:
I think we can add this field back if and when code_typed actually needs it, if and only if it makes sense at that point.